packages/gitbook/src/components/Footer/Footer.tsx

```diff
@@ -17,125 +19,123 @@ export function Footer(props: {
+                    )}
+                </div>
+
+                {/* Mode Switcher */}
                 <div
                     className={tcls(
-                        'flex',
-                        'flex-col',
-                        CONTAINER_STYLE,
-
-                        'py-6',
-                        'gap-6',
-                        'md:flex-row',
-                        'md:gap-10',
-
-                        //override CONTAINER_STYLE px
-                        'px-0',
-                        'sm:px-0',
+                        'ml-auto',
+                        'order-2 lg:order-4',
+                        customization.footer.groups?.length < 3 &&
+                            customization.footer.logo == undefined &&
+                            'sm:order-4',
+                        'xl:basis-56',
                     )}
                 >
-                    {' '}
-                    {customization.footer.logo ? (
-                        <div
-                            className={tcls(
-                                'pt-8',
-                                'border-t',
-                                'border-dark/2',
-                                'dark:border-light/1',
-                                'px-4',
-                                'sm:px-6',
-                                'md:px-0',
-                                'md:pr-10',
-                                'md:border-r',
-                                'md:border-t-0',
-                                'md:pt-0',
-                            )}
-                        >
-                            <Image
-                                alt="Logo"
-                                sources={{
-                                    light: {
-                                        src: customization.footer.logo.light,
-                                    },
-                                    dark: customization.footer.logo.dark
-                                        ? {
-                                              src: customization.footer.logo.dark,
-                                          }
-                                        : null,
-                                }}
-                                priority="lazy"
-                                style={['h-auto', 'w-full', 'max-w-12', 'object-contain']}
-                                sizes={[
-                                    {
-                                        width: 320,
-                                    },
-                                ]}
-                            />
+                    {customization.themes.toggeable && (
+                        <div className="flex items-center justify-end">
+                            <React.Suspense fallback={null}>
+                                <ThemeToggler />
+                            </React.Suspense>
                         </div>
-                    ) : null}
+                    )}
```

Comment:
I'd keep the pattern of `something ? <jsx/> : null` instead of `something && <jsx/>`, the latter can cause the UI to return a `null` string.